allennlp/commands/train.py

```diff
@@ -225,6 +225,9 @@ def train_model(
     training_util.create_serialization_dir(params, serialization_dir, recover, force)
     params.to_file(os.path.join(serialization_dir, CONFIG_NAME))
 
+    include_in_archive = params["trainer"].get("include_in_archive", None)
```

Comment:
Hey @eladsegal, I'm having trouble understanding this.

You have made `include_in_archive` a parameter of the `Trainer` class here, but it doesn't look like you actually added it to the `Trainer` class, nor does it seem like it should be a parameter of the `Trainer` class since the `Trainer` class doesn't seem to need it. This will probably cause `allennlp train` to fail when `include_in_archive` is used because it will be passed to the `Trainer`, resulting in `TypeError: Unexpected argument` or something.

Instead, it either needs to be a top-level field in `params` or a command-line flag `--include-in-archive`.

We should also add a test to `tests/commands/train_test.py`.

Reply:
Instead of adding a test in `train_test.py`, I changed the test in `archival_test.py` to be like the rest of the tests which test archival via calling `train_model`.
It is not calling `archive_model` directly to use `include_in_archive`, but adds it to the params.